examples/webview2/src/main.rs

```diff
@@ -0,0 +1,622 @@
+                    if let Some(op) = op.as_ref() {
+                        tx.send(op.GetResults()?).expect("send over mpsc channel");
+                    }
+                    Ok(())
+                }))?;
+
+            wait_with_pump(rx)
+        }?;
+
+        let size = get_window_size(parent);
+        let mut client_rect = RECT::default();
+        unsafe { WindowsAndMessaging::GetClientRect(parent, &mut client_rect) };
+        controller.SetBounds(Rect {
+            X: 0f32,
+            Y: 0f32,
+            Width: size.cx as f32,
+            Height: size.cy as f32,
+        })?;
+        controller.SetIsVisible(true)?;
+
+        let webview = controller.CoreWebView2()?;
+
+        if !debug {
+            let settings = webview.Settings()?;
+            settings.SetAreDevToolsEnabled(false)?;
+            settings.SetAreDefaultContextMenusEnabled(false)?;
+        }
+
+        let bindings: Arc<Mutex<HashMap<String, Box<dyn FnMut(u64, Vec<Value>)>>>> =
+            Arc::new(Mutex::new(HashMap::new()));
+        let bindings_ref = bindings.clone();
+        let token = webview.WebMessageReceived(TypedEventHandler::<
+            CoreWebView2,
+            CoreWebView2WebMessageReceivedEventArgs,
+        >::new(move |_sender, args| {
+            if let Some(args) = args {
+                if let Ok(message) = String::from_utf16(args.WebMessageAsJson()?.as_wide()) {
+                    if let Ok(value) = serde_json::from_str::<InvokeMessage>(&message) {
+                        let mut bindings = bindings_ref.lock().expect("lock bindings");
+                        if let Some(f) = bindings.get_mut(&value.method) {
+                            (*f)(value.id, value.params);
+                        }
+                    }
+                }
+            }
+            Ok(())
+        }))?;
+
+        if let Some(frame) = frame.as_ref() {
+            *frame.size.lock().expect("lock size") = size;
+        }
+
+        let (tx, rx) = mpsc::channel();
+        let rx = Arc::new(rx);
+        let thread_id = unsafe { SystemServices::GetCurrentThreadId() };
+
+        let webview = WebView {
+            controller: Arc::new(controller),
+            webview: Arc::new(webview),
+            tx,
+            rx,
+            thread_id,
+            token,
+            bindings,
+            frame,
+            parent: Arc::new(parent),
+            url: Arc::new(Mutex::new(HString::new())),
+        };
+
+        // Inject the invoke handler.
+        webview
+            .init(r#"window.external = { invoke: s => window.chrome.webview.postMessage(s) };"#)?;
+
+        if webview.frame.is_some() {
+            WebView::set_window_webview(parent, Box::new(webview.clone()));
+        }
+
+        Ok(webview)
+    }
+
+    pub fn run(self) -> Result<()> {
+        let webview = self.webview.as_ref();
+        let url = self.url.lock().expect("lock url").clone();
+        let (tx, rx) = mpsc::channel();
+
+        if url.len() > 0 {
```

Comment:
```suggestion
        if !url.is_empty() {
```

This is a bit more idiomatic.

Reply:
Yup. For some reason rust-analyzer didn't include that in the suggestions at the time, maybe I had a parse error elsewhere.